### PR TITLE
Use the VS 2015 libraries for MSVC 14

### PR DESCRIPTION
**Pull request by Furkanzmc**
When compiling with MSVC 14, the linker cannot find `libpng-2015.lib`, `libtiff-2015.lib`, `libjpeg-2015.lib` and `libglfw-2015.lib`. This commit fixes that problem. I haven't contributed to the CMake files in cocos2dx before so I'm not really sure If this is in the right format or, really, the best way to do it but it works.

### CODE REVIEW

File: cmake/Modules/CocosUsePrebuiltLibs.cmake [hunk 1]
```diff
@@ -4,7 +4,11 @@
 
 set(_chipmunk_inc chipmunk/chipmunk.h)
 set(_chipmunk_inc_paths include)
-set(_chipmunk_libs chipmunk libchipmunk)
+if (${MSVC_VERSION} STREQUAL "1900")
+    set(_chipmunk_libs chipmunk libchipmunk-2015)
```
Comment (minggo) [inline cmake/Modules/CocosUsePrebuiltLibs.cmake]: where is libchipmunk-2015?

Comment (Furkanzmc) [inline cmake/Modules/CocosUsePrebuiltLibs.cmake]: I don't understand what you mean. It is on line 7. Can you clarify?

Comment (minggo) [inline cmake/Modules/CocosUsePrebuiltLibs.cmake]: Sorry, i found it. There is libchipmunk-2015 in external.

**Comment by minggo** [comment]
Please fix compiling error:

```
CMake Error at cmake/Modules/CocosUsePrebuiltLibs.cmake:7 (if):
  if given arguments:

    "STREQUAL" "1900"

  Unknown arguments specified
Call Stack (most recent call first):
  CMakeLists.txt:94 (include)


-- Configuring incomplete, errors occurred!
See also "/linux_slave/workspace/ccs-pr/node/linux/linux-build/CMakeFiles/CMakeOutput.log".
cmake generate error
```

**Comment by Furkanzmc** [comment]
I'm not familiar with Windows Store and Phone applications, should I check for those platforms too? Is checking just for WINDOWS enough?

**Comment by minggo** [comment]
I am not sure. Most people just use VS to build apps. And could you please tell me how to reproduce the issue?

**Comment by Furkanzmc** [comment]
If you mean the initial issue, when using MSVC 14 the linker looks for libraries with `-2015` suffix but cannot find it. I'm much familiar with Cocos2dx's CMake files so I can't tell the root cause as I couldn't figure out where that `-2015` suffix is added :smile: But I found the `CocosUsePrebuiltLibs.cmake` file and just made the changes there to see if it works. And it did.
The second issue (Linux compilation issue) happens because of how `if` works in CMake. `if` treats `MSVC_VERSION` as a variable and it gives and error when it cannot find it or it is empty.

Since we are at it, is VS 2013 still supported on cocos2dx v3.12? because I get errors because the linker cannot find `version.lib` and whole lot of other errors because of the C implementation of the compiler. Apparently, VS 2015 changed its underlying C implemention so I thought maybe cocos2dx is using VS 2015 lib files for VS 2013 compiler too.

**Comment by minggo** [comment]
yep, VS2013 is still supports. Did you use command line to compile for windows?

**Comment by Furkanzmc** [comment]
No, I used Qt Creator with the default CMakeLists file.
And here's the command Qt Creator uses: `cmake --build --target-all`

**Comment by Furkanzmc** [comment]
These are the errors I get while compiling v3.12 with VS 2013

```
libeay32.lib(gost_eng.obj) : error LNK2001: unresolved external symbol ___acrt_iob_func
libeay32.lib(cryptlib.obj) : error LNK2001: unresolved external symbol ___acrt_iob_func
libeay32.lib(pem_lib.obj) : error LNK2001: unresolved external symbol ___acrt_iob_func
libeay32.lib(ui_openssl.obj) : error LNK2001: unresolved external symbol ___acrt_iob_func
libeay32.lib(gost_eng.obj) : error LNK2001: unresolved external symbol ___stdio_common_vfprintf
libeay32.lib(pem_lib.obj) : error LNK2001: unresolved external symbol ___stdio_common_vfprintf
libeay32.lib(ui_openssl.obj) : error LNK2001: unresolved external symbol ___stdio_common_vfprintf
libeay32.lib(e_cswift.obj) : error LNK2001: unresolved external symbol ___stdio_common_vsprintf
libeay32.lib(cryptlib.obj) : error LNK2001: unresolved external symbol ___stdio_common_vsprintf
libeay32.lib(dso_win32.obj) : error LNK2001: unresolved external symbol ___stdio_common_vsprintf
libeay32.lib(v3_utl.obj) : error LNK2001: unresolved external symbol ___stdio_common_vsscanf
```

**Comment by mirth** [comment]
Still can reproduce in Microsoft Visual Studio 2017. I created new project via 'cocos new' and generated msvc project with cmake.

**Comment by Furkanzmc** [comment]
@mirth This commit only fixes the CMake files. So I don't think it will be of any help for  Visual Studio projects.

**Comment by mirth** [comment]
@Furkanzmc but you've said that you've been compiling with  MSVC 14? explain please

**Comment by Furkanzmc** [comment]
@mirth You mentioned the IDE name, not the compiler. So I just assumed you were using the Visual Studio project file. Is that not the case? Sorry.

**Comment by mirth** [comment]
i generate visual studio project file with cmake though. can you give a link how to compile cocos project without ide in windows?

**Comment by Furkanzmc** [comment]
In my experience, generating the Visual Studio project file from CMake with Cocos2dx is not the way to go. Cocos2dx already has Visual Studio projects. And they work. I suggest you use those. Or you could directly use CMake to compile. Personally, I prefer Qt Creator over Visual Studio so I'm using CMake. I'm not sure if Visual Studio has CMake support but a quick Google search gave me [this](https://channel9.msdn.com/Events/Visual-Studio/Visual-Studio-2017-Launch/T138).

**Comment by mirth** [comment]
But how i should i maintain both win and os x build? So there will be two separate project for vs and code?

**Comment by Furkanzmc** [comment]
Are you developing for mobile or desktop?

If you are only developing for desktop, then you can use CMake for both Windows and macOS. If you are developing for mobile, you still need to modify the platform specific project files I believe. You can also use the [Cocos2dx's command line tool](http://www.cocos2d-x.org/wiki/Cocos_compile#iOS-Arguments) to build.

**Comment by mirth** [comment]
For mobile. But i want to have build for desktop too. What the option for CMake on Windos what target/compiler?

**Comment by Furkanzmc** [comment]
I'm not sure If I understand the question correctly but I'm assuming you are asking for which compiler to use with CMake.

You can use MSVC on Windows.

**Pull request merged**